683 nft add ipfs http andd aeweb url

Esquimor opened the pull request:
# Description

Add the possibility to add nft by using an url

Fixes #683 (issue)

## Type of change

- New feature (non-breaking change which adds functionality)
- Refactorization

## Material used:

- Android (Smartphone/Tablet)

## How Has This Been Tested?

Manually.

## Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

## Useful info for the reviewer:

I have change a lot of file. If you want an explanation: tell me.

Also, I put this pull request in draft. The url and aeweb is not done yet. But there are very easy to do.
So you can see the architecture of the pull request to add comments.

Also, I need some image.

## To do:

- [x] Url
- [x] Aeweb
- [x] Change Image
- [x] Add Label

redDwarf03 committed:
Prepare web version to security audit

Commit:
Refactor grid for NFT import category

Commit:
Refactor grid for NFT import category

Commit:
Refactor grid for NFT import category

Commit:
683-Verify url ipfs

Commit:
683-Verify url ipfs

Commit:
683-Verify url ipfs

Commit:
Create Ipfs nft

Commit:
Create Ipfs nft

Commit:
Create Ipfs nft

Commit:
Show ipfs NFT in user library

Commit:
Show ipfs NFT in user library

Commit:
Refacto Code for Image and Form

Commit:
Refacto Code for Image and Form

redDwarf03 commented on lib/ui/views/nft_creation/layouts/components/card_cateogry.dart:
```diff
@@ -0,0 +1,32 @@
+/// SPDX-License-Identifier: AGPL-3.0-or-later
```
Typo for the name of file cateogry -> category

redDwarf03 commented on lib/ui/views/nft_creation/bloc/provider.dart:
```diff
@@ -491,17 +502,46 @@ class NftCreationFormNotifier extends FamilyNotifier<NftCreationFormState,
+      ...state.properties,
+      NftCreationFormStateProperty(
+        propertyName: 'content',
+        propertyValue: {'ipfs_url': uri},
```
Accord to the AEIP 9 : 'ipfs' instead of 'ipfs_url'

redDwarf03 commented on lib/ui/views/nft_creation/bloc/provider.dart:
Change to ``aeip: [2, 9],`` line 151

redDwarf03 commented on lib/ui/views/nft/layouts/components/nft_item_ipfs.dart:
```diff
@@ -0,0 +1,64 @@
+                      borderRadius: BorderRadius.circular(15),
+                      child: ImageNetworkSageWidgeted(
+                        url: ipfsFormattedUrl,
+                        errorMessage: 'Oops! The IPFS URL is not available.',
```
Externalize message to localization

redDwarf03 commented on lib/ui/views/nft_creation/layouts/components/nft_creation_process_import_tab_template_form.dart:
```diff
@@ -0,0 +1,135 @@
+        child: Column(
+          children: <Widget>[
+            SheetHeader(
+              title: localizations.addAccount,
```
wrong label

redDwarf03 commented on lib/ui/views/nft_creation/layouts/components/nft_creation_process_import_tab_ipfs.dart:
```diff
@@ -0,0 +1,37 @@
+          ),
+        );
+      },
+      text: localizations.nftAddImportPhoto,
```
wrong label

redDwarf03 reviewed (commented):
with macos, i haven't all menus
<img width="358" alt="Capture d’écran 2023-03-13 à 23 23 12" src="https://user-images.githubusercontent.com/68340133/224848408-d912cd7d-1d34-4448-a36a-341907bd9d92.png">

Chralu committed:
:bug: RPC events unicity

Successive events with same value are not emitted.

Chralu committed:
:bug: RPC subscription returns current value.

On RPC subscription, a notification with the current value is immediately sent.

redDwarf03 committed:
Add get_current_account Command

Chralu committed:
:bug: RPC error message is not obfuscated.

Chralu committed:
:bug: Subscription fixed.

Chralu committed:
:bug: Subscription fixed.

redDwarf03 committed:
Add currentAccount subscription

redDwarf03 committed:
fix sub

Commit:
683 Add link for http url

Commit:
683 Add link for http url

Commit:
683 Add link for http url

Commit:
683-Add Labels

Commit:
683-Add Labels

Commit:
683-Typo error

Commit:
683-Typo error

Commit:
683-Typo error

redDwarf03 committed:
add handler

Chralu committed:
[RPC] Upgrade subscription notification protocol.

Chralu committed:
[RPC] SubCurrentAccount only shows account Identity.

Chralu committed:
:bug: fix RPC current account.

Chralu committed:
:bug: ensures RPCCommandReceiver never disposed.

Chralu committed:
:bug: fix accountNotifier on removed wallet.

Chralu committed:
:feature: RPC returns specific failure when no wallet is setup.

Chralu committed:
:feature: RPC SubAccount returns failure if requested account does not exist.

Commit:
683-Add AEWeb and image

Commit:
683-Add AEWeb and image

Commit:
683-Add AEWeb and image

redDwarf03 committed:
update pointycastle to 3.7.0

redDwarf03 committed:
fix pubspec.lock

redDwarf03 committed:
update dependencies

redDwarf03 committed:
fix error with flutter web

redDwarf03 committed:
update libdart dependency

Commit:
Merge branch '683-NFT-add-IPFS-http-andd-aeweb-url' of https://github.com/Esquimor/archethic-wallet into 683-NFT-add-IPFS-http-andd-aeweb-url

Commit:
Merge branch '683-NFT-add-IPFS-http-andd-aeweb-url' of https://github.com/Esquimor/archethic-wallet into 683-NFT-add-IPFS-http-andd-aeweb-url

Commit:
683-Fix merge conflict

Esquimor commented:
Move to https://github.com/archethic-foundation/archethic-wallet/pull/714
